internal/oidc/oauth2.go

```diff
@@ -0,0 +1,152 @@
+package oidc
+
+import (
+	"time"
+
+	"github.com/fasthttp/router"
+	"github.com/ory/fosite"
+	"github.com/ory/fosite/compose"
+	"github.com/ory/fosite/handler/openid"
+	"github.com/ory/fosite/storage"
+	"github.com/ory/fosite/token/jwt"
+
+	"github.com/authelia/authelia/internal/configuration/schema"
+	"github.com/authelia/authelia/internal/middlewares"
+	"github.com/authelia/authelia/internal/utils"
+)
+
+// NewStore creates a new OIDC store.
+func NewStore(config *schema.OpenIDConnectConfiguration) *storage.MemoryStore {
+	clients := make(map[string]fosite.Client)
+
+	for _, v := range config.Clients {
+		clients[v.ID] = &fosite.DefaultClient{
+			ID:            v.ID,
+			Secret:        []byte(v.Secret),
+			RedirectURIs:  v.RedirectURIs,
+			ResponseTypes: v.ResponseTypes,
+			GrantTypes:    v.GrantTypes,
+			Scopes:        v.Scopes,
+		}
+	}
+
+	return &storage.MemoryStore{
```

Comment:
We definitely need to replace that as soon as possible. It might be ok to have that in beta but in v1, we should definitely have an on-disk option at least to recover after a restart of Authelia.

Reply:
I'm fine with a v1 implementation of it, as long as we aim to shortly after add that option. It would be breaking for anyone engaging in distributed workloads.